app/src/main/java/com/github/orkest/View/feed/FeedActivity.kt

```diff
@@ -0,0 +1,223 @@
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.orkest.Model.Post
+import com.github.orkest.Model.Profile
+import com.github.orkest.Model.Song
+import com.github.orkest.Model.User
+import com.github.orkest.R
+import com.github.orkest.View.search.SearchUserView
+
+
+/**
+ * Composable function for the feed screen
+ * Represents the view of the MVVM pattern
+ */
+@Composable
+fun FeedActivity(){
+    //Add a list of posts
+    val user = User("Username", profile = Profile(username = "Username",
+                                            profilePictureId = R.drawable.profile_picture))
+    val rudeBoySong = Song("Rude Boy", "Rihanna", "Rated R",
+                "link", R.drawable.album_cover)
+    val post = Post(user,"Post Description", rudeBoySong, 0, ArrayList())
+
+    var listPosts by remember { mutableStateOf(mutableListOf(post,post,post,post)) }
+    LazyColumn(modifier = Modifier
+        .fillMaxSize()
+        .background(Color.LightGray)) {
+        items(listPosts) { post ->
+            Post(post = post)
+        }
+    }
+}
+
+/**
+ * Composable function to display a post, can be reused for the profile page
+ * @param post the post to display
+ */
+@Composable
+fun Post(post: Post){
+
+    Row(modifier = Modifier
+        .padding(start = 25.dp, top = 20.dp)
+        .clip(shape = RoundedCornerShape(20.dp))
+        .background(Color.DarkGray)){
+
+        Column {
+            // Display the user profile pic
+            UserProfilePic(post.user.profile.profilePictureId)
+            //Display the reaction buttons
+            Reaction()
+        }
+
+        Column(modifier = Modifier.padding(10.dp, top = 10.dp, end = 10.dp)) {
+            //Add the user's username
+            UserUsername(post.user.username)
+            // Display the post's content
+            PostDescription(post.postDescription)
+            Spacer(modifier = Modifier.height(10.dp))
+            // Display the post's song
+            SongCard(post.song)
+            Spacer(modifier = Modifier.height(10.dp))
+        }
+    }
+}
+
+@Composable
+private fun UserUsername(username: String){
+    //Add the user's username
+    Text(text = username, fontSize = 14.sp, fontWeight = FontWeight.Bold,
+        color = Color.White,
+        modifier = Modifier
+          //  .padding(start = 10.dp)
+            .clickable {  })
+}
+
+@Composable
+private fun UserProfilePic(profilePicId : Int){
```

Comment:
Same remark for UserProfilePic: there's a single type of profile thus the profile pic can only belong to a user